src/extension.ts

```diff
@@ -1,20 +1,102 @@
 
-    const connection = new Connection(context)
-    await connection.load()
+    const treeProvider = new InfluxDBTreeProvider(context)
+    context.subscriptions.push(
+        vscode.window.registerTreeDataProvider(
+            'influxdb', treeProvider
+        )
+    )
+    context.subscriptions.push(
+        vscode.commands.registerCommand(
+            'influxdb.refresh', () => { treeProvider.refresh() }
+        )
+    )
+    context.subscriptions.push(
+        vscode.commands.registerCommand(
+            'influxdb.runQuery',
+            // XXX: rockstar (26 Aug 2021) - This should really live in a controller of some sort,
+            // but the current abstractions make it hard to see how that architecture should be.
+            async () => {
+                const { activeTextEditor } = vscode.window;
+                if (!activeTextEditor) {
+                    return
+                }
+                let query = ''
+                // Is this functionality actually important?
+                if (activeTextEditor.selection.isEmpty) {
+                    query = activeTextEditor.document.getText()
+                } else {
+                    query = activeTextEditor.document.getText(activeTextEditor.selection)
+                }
+                if (!query) {
+                    vscode.window.showWarningMessage('No flux file selected')
+                    return
+                }
+                try {
+                    const connections = context.globalState.get<{ [key : string] : IConnection; }>(InfluxDBConnectionsKey) || {}
+                    const connection = Object.values(connections).filter((item : IConnection) => item.isActive)[0]
+                    const queryApi = new InfluxDB({ url: connection.hostNport, token: connection.token }).getQueryApi(connection.org)
+                    const results = await QueryResult.run(queryApi, query)
+                    const tableView = new TableView(context)
+                    tableView.show(results, connection.name)
+                } catch (error) {
+                    vscode.window.showErrorMessage(error.message)
+                    console.error(error)
+                }
+            }
+        )
+    )
+    context.subscriptions.push(
+        vscode.commands.registerCommand(
+            'influxdb.addConnection',
+            async () => {
+                const addConnectionView = new ConnectionView(context, treeProvider)
+                await addConnectionView.create()
+            }
+        )
+    )
+    context.subscriptions.push(
+        vscode.commands.registerCommand(
+            'influxdb.removeConnection',
+            async (node : Connection) => {
+                node.removeConnection(node)
+            }
+        )
+    )
+    context.subscriptions.push(
+        vscode.commands.registerCommand(
+            'influxdb.editConnection',
+            async (node : Connection) => {
+                node.editConnection(node)
```

Comment:
Why is node passed in as an argument and is the `this/self` value for the method?

Reply:
Ah yes, holdover from the transition. The argument isn't even used, so I removed it. Thanks for catching that.